bundles/org.openhab.binding.iaqualink/src/main/java/org/openhab/binding/iaqualink/internal/handler/IAqualinkHandler.java

```diff
@@ -0,0 +1,450 @@
+
+    /**
+     * Local cache of iAqualink states
+     */
+    private Map<String, State> stateMap = Collections.synchronizedMap(new HashMap<>());
+
+    /**
+     * Our poll rate
+     */
+    private int refresh;
+
+    /**
+     * fixed API key provided by iAqualink clients (Android, IOS), unknown if this will change in the future.
+     */
+    @Nullable
+    private String apiKey;
+
+    /**
+     * Optional serial number of the pool controller to connect to, only useful if you have more then one controller
+     */
+    @Nullable
+    private String serialNumber;
+
+    /**
+     * Server issued sessionId
+     */
+    @Nullable
+    private String sessionId;
+
+    /**
+     * When we first connect we will dynamically create channels based on what the controller is configured for
+     */
+    private boolean firstRun;
+
+    /**
+     * Future to poll for updated
+     */
+    @Nullable
+    private ScheduledFuture<?> pollFuture;
+
+    /**
+     * The client interface to the iAqualink Service
+     */
+    private IAqualinkClient client;
+
+    /**
+     * Constructs a new {@link IAqualinkHandler}
+     *
+     * @param thing
+     * @param httpClient
+     */
+    public IAqualinkHandler(Thing thing, HttpClient httpClient) {
+        super(thing);
+        client = new IAqualinkClient(httpClient);
+    }
+
+    @Override
+    public void initialize() {
+        configure();
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposed.");
+        clearPolling();
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        // clear our cached value so the new channel gets updated on the next poll
+        stateMap.remove(channelUID.getId());
+    }
+
+    @SuppressWarnings("null") // stateMap.remove can return null, which confuses the compiler
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("handleCommand channel: {} command: {}", channelUID, command);
+
+        if (getThing().getStatus() != ThingStatus.ONLINE) {
+            logger.warn("Controller is NOT ONLINE and is not responding to commands");
+            return;
+        }
+
+        clearPolling();
+
+        String channelName = channelUID.getIdWithoutGroup();
+        // remove the current state to ensure we send an update
+        State oldState = stateMap.remove(channelUID.getAsString());
+        try {
+            if (command instanceof RefreshType) {
+                stateMap.clear();
+            } else if (channelName.startsWith("onetouch_")) {
+                if (oldState == null || !oldState.toString().equals(command.toString())) {
+                    client.oneTouchCommand(serialNumber, sessionId, "set_" + channelName);
+                }
+            } else if (channelName.endsWith("heater") || channelName.endsWith("pump")) {
+                if (oldState == null || !oldState.toString().equals(command.toString())) {
```

Comment:
What about using a helper method to compare `State` and `Command` values and maybe consider not comparing on to String but using something else, if it's just the regular enums and you want to match on equal values you can just use ==

Reply:
Yes, that is a good idea.